include/SDL3/SDL_filesystem.h

```diff
@@ -236,6 +236,102 @@ typedef enum
  */
 extern DECLSPEC char *SDLCALL SDL_GetUserFolder(SDL_Folder folder);
 
+
+/* Abstract filesystem interface */
+
+typedef enum SDL_PathType
+{
+    SDL_PATHTYPE_FILE, /**< a normal file */
+    SDL_PATHTYPE_DIRECTORY, /**< a directory */
+    SDL_PATHTYPE_OTHER /**< something completely different like a device node (not a symlink, those are always followed) */
+} SDL_PathType;
+
+/* SDL file timestamps are 64-bit integers representing seconds since the Unix epoch (Jan 1, 1970) */
+typedef Sint64 SDL_FileTimestamp;
+
+typedef struct SDL_PathInfo
+{
+    SDL_PathType type;              /* the path type */
+    Uint64 size;                    /* the file size in bytes */
+    SDL_FileTimestamp create_time;  /* the time when the path was created */
+    SDL_FileTimestamp modify_time;  /* the last time the path was modified */
+    SDL_FileTimestamp access_time;  /* the last time the path was read */
+} SDL_PathInfo;
+
+/**
+ * Create a directory.
+ *
+ * \param path the path of the directory to create
+ * \returns 0 on success or a negative error code on failure; call
+ *          SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0.
+ */
+extern DECLSPEC int SDLCALL SDL_CreateDirectory(const char *path);
```

Comment:
Will this be recursive by chance? Not a dealbreaker but would be cool to have SDL handle the loop here!

Reply:
Yeah, I'm throwing a TODO in SDL_CreateDirectory()